lxd/storage_btrfs.go

```diff
@@ -1319,82 +1329,70 @@ func (s *storageBtrfs) btrfsPoolVolumeQGroupUsage(subvol string) (int64, error)
 	return -1, fmt.Errorf("Unable to find current qgroup usage")
 }
 
-func (s *storageBtrfs) btrfsPoolVolumeDelete(subvol string) error {
+func btrfsSubVolumeDelete(subvol string) error {
 	// Attempt (but don't fail on) to delete any qgroup on the subvolume
-	qgroup, err := s.btrfsPoolVolumeQGroup(subvol)
+	qgroup, err := btrfsSubVolumeQGroup(subvol)
 	if err == nil {
-		output, err := exec.Command(
+		exec.Command(
 			"btrfs",
 			"qgroup",
 			"destroy",
 			qgroup,
-			subvol).CombinedOutput()
-
-		if err != nil {
-			s.log.Warn(
-				"subvolume qgroup delete failed",
-				log.Ctx{"subvol": subvol, "output": string(output)},
-			)
-		}
+			subvol).Run()
 	}
 
 	// Attempt to make the subvolume writable
 	exec.Command("btrfs", "property", "set", subvol, "ro", "false").CombinedOutput()
 
 	// Delete the subvolume itself
-	output, err := exec.Command(
+	err = exec.Command(
 		"btrfs",
 		"subvolume",
 		"delete",
 		subvol,
-	).CombinedOutput()
+	).Run()
 
-	if err != nil {
-		s.log.Warn(
-			"subvolume delete failed",
-			log.Ctx{"subvol": subvol, "output": string(output)},
-		)
-	}
 	return nil
```

Comment:
I'm tempted to say that even the qgroup stuff should be fatal if it fails. We only run that code if we detected a qgroup exists, so if we fail to remove it, there's something wrong.